internal/gengapic/genrest.go

```diff
@@ -1126,3 +1155,48 @@ func (g *generator) unaryRESTCall(servName string, m *descriptor.MethodDescripto
 	g.imports[outSpec] = true
 	return nil
 }
+
+func (g *generator) restCallOptions(serv *descriptor.ServiceDescriptorProto, servName string) {
+	p := g.printf
+
+	// defaultCallOptions
+	c := g.grpcConf
+
+	methods := append(serv.GetMethod(), g.getMixinMethods()...)
+
+	// read retry params from gRPC ServiceConfig
+	p("func default%[1]sRESTCallOptions() *%[1]sCallOptions {", servName)
+	p("  return &%sCallOptions{", servName)
+	for _, m := range methods {
+		sFQN := g.fqn(g.descInfo.ParentElement[m])
+		mn := m.GetName()
+		p("%s: []gax.CallOption{", mn)
+		if rp, ok := c.RetryPolicy(sFQN, mn); ok && rp != nil {
+			p("gax.WithRetry(func() gax.Retryer {")
+			p("  return gax.OnHTTPCodes(gax.Backoff{")
+			// this ignores max_attempts
+			p("    Initial:    %d * time.Millisecond,", conf.ToMillis(rp.GetInitialBackoff()))
+			p("    Max:        %d * time.Millisecond,", conf.ToMillis(rp.GetMaxBackoff()))
+			p("    Multiplier: %.2f,", rp.GetBackoffMultiplier())
+			p("	 },")
+
+			rc := rp.GetRetryableStatusCodes()
+			for ndx, c := range rc {
```

Comment:
Is it possible for rc to be nil here? Do we first need to check if there are codes before adding the retry?

Reply:
It is _possible_, but would be considered a misconfiguration. If a method should not be retried, it shouldn't have a retry policy at all. I will add a check though, no point in not :)